From f33a389a48a9769e90c4d6cb7a515d123c815d05 Mon Sep 17 00:00:00 2001
From: Peter Chen <peter.chen@freescale.com>
Date: Tue, 23 Jun 2015 16:04:57 +0800
Subject: [PATCH 0151/5242] MLK-11151 usb: chipidea: otg: drive vbus off if
 fsm.drv_vbus is true

commit  3d257643e07b3c2642375bd0b8f4fe7180b8e33f from
https://source.codeaurora.org/external/imx/linux-imx.git

The current condition may not true even we have driven vbus on already.
The vbus drive operation may be async, eg through i2c gpio expendor,
so the vbus may not be valid when we unload usb module.

We meet this issue when we do fast load/unload usb module test.

Signed-off-by: Peter Chen <peter.chen@freescale.com>
(cherry picked from commit c0cdff8472c38e0fc39c597148d5534a5e9cc9b1)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/chipidea/otg_fsm.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/chipidea/otg_fsm.c b/drivers/usb/chipidea/otg_fsm.c
index 4a34fd2..1249925 100644
--- a/drivers/usb/chipidea/otg_fsm.c
+++ b/drivers/usb/chipidea/otg_fsm.c
@@ -938,7 +938,7 @@ void ci_hdrc_otg_fsm_remove(struct ci_hdrc *ci)
 	ci->enabled_otg_timer_bits = 0;
 
 	/* Turn off vbus if vbus is on */
-	if (ci->fsm.a_vbus_vld)
+	if (ci->fsm.drv_vbus)
 		otg_drv_vbus(&ci->fsm, 0);
 
 	sysfs_remove_group(&ci->dev->kobj, &inputs_attr_group);
-- 
1.7.9.5

